wa/workloads/pcmark/uiauto/app/src/main/java/com/arm/wa/uiauto/pcmark/UiAutomation.java

```diff
@@ -0,0 +1,138 @@
+/*    Copyright 2014-2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arm.wa.uiauto.pcmark;
+
+import android.os.Bundle;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+import com.arm.wa.uiauto.ActionLogger;
+import android.util.Log;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    private int networkTimeoutSecs = 30;
+    private long networkTimeout =  TimeUnit.SECONDS.toMillis(networkTimeoutSecs);
+    public static String TAG = "UXPERF";
+
+    @Before
+    public void initialize(){
+        initialize_instrumentation();
+    }
+
+    @Test
+    public void setup() throws Exception{
+        setScreenOrientation(ScreenOrientation.NATURAL);
+        loadBenchmarks();
+        installBenchmark();
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        runBenchmark();
+    }
+
+    @Test
+    public void teardown() throws Exception{
+        unsetScreenOrientation();
+    }
+
+    //Swipe to benchmarks and back to initialise the app correctly
+    private void loadBenchmarks() throws Exception {
+        UiObject title = 
+            mDevice.findObject(new UiSelector().resourceId("android:id/action_bar_spinner")
```

Comment:
This is still failing to be found for me, I've attached the UI dump for comparison. 
[setup-error.tar.gz](https://github.com/ARM-software/workload-automation/files/1995503/setup-error.tar.gz)

Reply:
This looks very different to the application I am running. 

From your dump it looks like you can see Install correctly on first launch. This isn't the same behaviour that we see on mobile devices. I will add in some logic to handle this better.

Reply:
Ah ok, I've been testing this with version 2.0.3716 (the latest available the play store) just in case this is different to the version you're using.